### ISSUE TITLE
More cask installation options should survive upgrade

### ISSUE THREAD

**Issue by matt-phylum** [Homebrew/brew#16994]
### Verification

- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### Provide a detailed description of the proposed feature

Homebrew should remember when casks are installed with `--no-quarantine` so it doesn't reenable quarantine when upgrading them. As it is, the `--no-quarantine` option is not very usable.

### What is the motivation for the feature?

The Ghidra project does not codesign important components of Ghidra (https://github.com/NationalSecurityAgency/ghidra/issues/1559#issuecomment-930110351). In practice, this means when you `brew install --cask ghidra` (https://formulae.brew.sh/cask/ghidra), you get a broken version of Ghidra which can't be used for binary analysis.

There is no real solution. Either you circumvent Gatekeeper by removing the quarantine attributes from the shipped binaries, or you circumvent Gatekeeper by building your own binaries without the quarantine attribute (by running unsigned scripts to build unsigned source code that is equally as trustworthy as the precompiled binaries).

The Homebrew project will not circumvent Gatekeeper in the cask (https://github.com/Homebrew/homebrew-cask/pull/169538#discussion_r1532529574).

As a workaround, you can install the package with `brew install --no-quarantine ghidra` (https://github.com/Homebrew/homebrew-cask/issues/137489#issuecomment-1345603369). However, Homebrew forgets that it's supposed to use `--no-quarantine` as soon as the installation completes. After a few weeks, `brew upgrade` breaks Ghidra, and attempting to open a new file in Ghidra results in repeated error dialogs from Mac OS telling you that you aren't allowed to run those executables and broken Ghidra functionality.

Homebrew does record `--appdir`, `--keyboard-layoutdir`, `--colorpickerdir`, etc into `.metadata/config.json`, but it does not record `--no-quarantine`.

### How will the feature be relevant to at least 90% of Homebrew users?

This is relevant to nearly 100% of users of similarly broken casks, but this is the only one I know of.

### What alternatives to the feature have been considered?

If this feature is not added, the Ghidra cask should be removed because it does not work. Maybe it could be replaced with a Ghidra formula because formulas turn unsigned source code into executable code without being constrained by Gatekeeper.

**Comment by MikeMcQuaid** [comment]
I don't think we should persist flags like this because that disable security features.

> If this feature is not added, the Ghidra cask should be removed because it does not work.

Moving to homebrew-cask and CC @Homebrew/cask for discussion.

My position: if literally nothing in this cask works at all for anyone without `--no-quarantine`: it should be removed. 

If some features are still useful without: it should stay, perhaps with some `caveats` added.
